### ISSUE TITLE
Use names as primary reference to pipelines in API

### ISSUE THREAD

**Issue by omarzouk** [kubeflow/pipelines#3360]
Pipeline names are unique per KFP cluster. Moreover, they are not editable. Which means names and uuids have a 1-to-1 relationship.

Using names as the primary identifier in the API would have a few benefits:

- CI/CD deployment usecase: for continuous deployment, it is an anti-pattern to have to store a uuid reference to a deployment in order to update it in consecutive runs. Today one can get around this by searching using the name, but if the relationship is 1-to-1 then its just an extra unnecessary step.
- Consistency with other tools: All resources and entities on kubernetes are referenced by name, all DAGs on airflow etc. are referenced by name, many of the existing workflows that users already have, expect to interact with the DAG and deployment environment via names rather than uuids.

**Comment by rmgogogo** [comment]
(need to keep eyes on IR design for this item)

**Comment by NikeNano** [comment]
> Pipeline names are unique per KFP cluster. Moreover, they are not editable. Which means names and uuids have a 1-to-1 relationship.

Names don't have to be unique, they are unique based upon the combination of `name` and `namespace`, we could use this as an composite key, but we would need the UUID otherwise as a Primary key. https://github.com/kubeflow/pipelines/blob/31429dc29616c87bd6c7aeeeac93ca9b779dd896/backend/src/apiserver/client_manager.go#L272 WDYT @omarzouk and @rmgogogo ?

**Comment by Bobgy** [comment]
Good point

**Comment by omarzouk** [comment]
@NikeNano Good point. Having a name+namespace as a key, or at least some utility function that lets you fetch pipelines using their combination is probably good enough.

Today my solution to work around this is to call the list function, and do a string search in the list to find the corresponding uuid and I'm worried that this behaviour would break in future versions.

**Comment by Bobgy** [comment]
Welcome contribution on this direction. Agree this will be useful

**Comment by difince** [comment]
Hi, Can I give this issue try .. probably by chaning just one endpoint as a start?

**Comment by Bobgy** [comment]
Sure, @difince!

Thank you for willing to take this!

**Comment by difince** [comment]
Hi @Bobgy I apologize for not being able to work on the issue for so long. I am back .. so hope soon I will have something to share/commit

**Comment by Bobgy** [comment]
No worries, no one is expected to work on anything. It's great that you are back!

**Comment by difince** [comment]
Hello @Bobgy Again I apologize for being inactive. 
In my understanding, I need to add two new endpoints - getPipelineByName and deletePipelineByName ?

**Comment by difince** [comment]
I have created a separate [issue](https://github.com/kubeflow/pipelines/issues/7003) for getPipelineByName and a [PR](https://github.com/kubeflow/pipelines/pull/7004) associated with it - to keep both the review process and comments addressing easier.

**Comment by difince** [comment]
As endpoint "getPipelineByName" already exists. Is it ok as a follow-up issue to work on adding another endpoint - `deletePipelineByName` ? @Bobgy @zijianjoy

**Comment by juliusvonkohout** [comment]
/close
due to inactivity
please /reopen if needed.
